pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -211,32 +210,38 @@ func (ctrl *Controller) deleteKubeletConfig(obj interface{}) {
 			return
 		}
 	}
-	if err := ctrl.cascadeDelete(cfg); err != nil {
-		utilruntime.HandleError(fmt.Errorf("couldn't delete object %#v: %v", cfg, err))
-	} else {
-		glog.V(4).Infof("Deleted KubeletConfig %s and restored default config", cfg.Name)
-	}
+	glog.V(4).Infof("Deleting KubeletConfig %s", cfg.Name)
+	ctrl.enqueueKubeletConfig(cfg)
```

Comment:
I think the overall approach work but if this is not as urgent (as it's not a regression) maybe we want to have a better mapping between KC and MC? Right now we have N KC -> 1 MC being re-generated everytime. We really want to have N KC -> 1 MC in order to make this MC-like and in order for the correct deletion of the MCs using finalizers